autoload/wiki/journal.vim

```diff
@@ -53,6 +53,43 @@ function! wiki#journal#go_to_month() abort " {{{1
   call wiki#url#parse('journal:' . l:date[:3] . '_m' . l:date[5:6]).open()
 endfunction
 
+" }}}1
+function! wiki#journal#make_index() " {{{1
+  let l:regex_days = '\d\{4}-\d\d-\d\d'
+  let l:entries = s:get_links_generic(l:regex_days, '%Y-%m-%d')
+
+  let l:sorted_entries = {}
+  for entry in entries
+    let [year, month, day] = split(entry, '-')
+    if has_key(sorted_entries, year)
+      let year_dict = sorted_entries[year]
+      if has_key(year_dict, month)
+        call add(year_dict[month], entry)
+      else
+        let year_dict[month] = [entry]
+      endif
+    else
+      let sorted_entries[year] = {month:[entry]}
+    endif
+  endfor
+
+  for year in reverse(sort(keys(sorted_entries)))
+    let l:month_dict = sorted_entries[year]
+    put ='# '.year
```

Comment:
Add spaces before/after the `.` (makes it easier to distinguish from dict members).